### PR TITLE
[PARSER] allow registry pattern  to add new parsers standalone from DMLC

### PR DESCRIPTION
**Pull request by tqchen**
@mli @hjk41

### CODE REVIEW

**Comment by tqchen** [comment]
@mli can you review and merge this?

File: include/dmlc/registry.h [hunk 1]
```diff
@@ -188,8 +188,8 @@ class FunctionRegEntryBase {
  */
 #define DMLC_REGISTRY_ENABLE(EntryType)                                 \
   template<>                                                            \
-  Registry<EntryType> *Registry<EntryType>::Get() {                     \
-    static Registry<EntryType> inst;                                    \
+  Registry<EntryType > *Registry<EntryType >::Get() {                   \
```
Comment (hjk41) [inline include/dmlc/registry.h]: why do we have blackspace after EntryType? is it a coding convention?

**Comment by hjk41** [comment]
looks good to me

File: include/dmlc/registry.h (hunk 1 above)
Comment (tqchen) [inline include/dmlc/registry.h]: EntryType could contain something that ends with ```>```, which I am not sure recognized by old c++ compiler(should be ok for c++11 compatible ones) 

```vector<vector<int>> ``` vs ```vector<vector<int> > ```

**Comment by tqchen** [comment]
Ideally this allow plug in of user customized parser into dmlc modules without actually putting code in dmlc codebase

**Comment by hjk41** [comment]
I see. Looks great.

On Tue, Dec 1, 2015 at 1:32 PM, Tianqi Chen <notifications@github.com>
wrote:

> Ideally this allow plug in of user customized parser into dmlc modules
> without actually putting code in dmlc codebase
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/dmlc-core/pull/75#issuecomment-160858191>.
>



-- 
HONG Chuntao
System Research Group
Microsoft Research Asia

**Pull request merged**